gitoxide-core/src/repository/log.rs

```diff
@@ -0,0 +1,152 @@
+use gix::bstr::{BStr, BString, ByteSlice};
+use gix::prelude::FindExt;
+use gix::ObjectId;
+
+pub fn log(mut repo: gix::Repository, out: &mut dyn std::io::Write, pathspec: BString) -> anyhow::Result<()> {
```

Comment:
Would `pathspec` not be optional? I think it should be.

Reply:
Definitely makes sense!